### DISCUSSION TITLE
A quoi sert GitHub ?

### DISCUSSION THREAD

**Discussion by LOQUITO77** [community/community#58932] [New to GitHub]
Bonjour , je m interroge , que puis je faire avec GitHub , à quoi sert il  ?

**Comment by Gustavonn07** [comment]
Bonjour! GitHub est une plateforme d'hébergement de code et de collaboration qui permet aux développeurs de travailler ensemble sur des projets logiciels. Voici certaines choses que vous pouvez faire avec GitHub :

1. Contrôle de version : GitHub utilise le système de contrôle de version Git, qui vous permet de suivre les modifications apportées au code source au fil du temps. Cela facilite la collaboration en équipe car plusieurs personnes peuvent travailler sur le même projet sans conflit.

2. Hébergement de référentiels : GitHub vous permet de créer des référentiels pour stocker vos projets logiciels. Cela signifie que vous pouvez enregistrer votre code source, votre documentation, vos fichiers de configuration et toute autre ressource liée au projet en un seul endroit.

3. Collaboration : GitHub permet à plusieurs personnes de collaborer facilement sur un projet. Vous pouvez inviter d'autres développeurs à collaborer sur votre référentiel, ce qui leur permet de contribuer au code, d'examiner les modifications, de suggérer des améliorations et de gérer les problèmes.

4. Gestion des problèmes : GitHub dispose de fonctionnalités intégrées de suivi des problèmes où vous pouvez créer des problèmes, les attribuer aux membres de l'équipe, suivre leur statut et discuter des solutions. Ceci est particulièrement utile pour suivre les bogues, les demandes de fonctionnalités et les tâches dans un projet.

5. Intégration continue : GitHub peut être intégré à des outils d'intégration continue tels que Travis CI ou Jenkins. Cela vous permet d'automatiser la création, les tests et le déploiement de votre projet à chaque fois qu'il y a un changement dans le référentiel.

6. Communauté et Open Source : GitHub abrite une large communauté de développeurs de logiciels. Vous pouvez découvrir des projets intéressants, contribuer à des projets open source, apprendre des autres développeurs et vous bâtir une réputation dans la communauté.

Ce ne sont là que quelques-unes des nombreuses choses que vous pouvez faire avec GitHub. Il s'agit d'une plate-forme polyvalente appréciée des développeurs de logiciels en raison de ses capacités de collaboration et de contrôle de version.

(Désolé pour les fautes, je ne parle pas français et j'ai dû traduire !)

**Comment by queenofcorgis** [comment]
Thanks @Gustavonn07 for chiming in! As a reminder, we take our [Code of Conduct](https://github.com/community/community/blob/0ca37cf594e438fd4aec3c7f3736b95e6b0b3567/CODE_OF_CONDUCT.md) very seriously and want to help ensure that everyone has a good experience free of antagonism and harassment. Unfortunately, we don’t currently have moderators for languages other than English. Until that changes, we need to ask that everyone use English here in the GitHub Community when posting. 

Check out this [check-in](https://github.com/orgs/community/discussions/42302#discussion-4683140) with links on how to choose your first project to contribute to, a blog with getting started resources, and a new user checklist.

Best of luck!

**Comment by leonden** [comment]
Bonjour ! GitHub est une plateforme de développement collaboratif basée sur Git, un système de contrôle de version très populaire. Voici quelques utilisations courantes de GitHub :

1. Gestion de projet : Vous pouvez utiliser GitHub pour héberger et gérer vos projets logiciels. Vous pouvez créer des référentiels (repositories) pour stocker votre code source, gérer les problèmes (issues) et les tâches, suivre les versions et collaborer avec d'autres développeurs.

2. Contrôle de version : GitHub utilise Git pour le contrôle de version, ce qui signifie que vous pouvez enregistrer l'historique complet des modifications apportées à votre code. Cela vous permet de revenir à des versions antérieures, de fusionner les modifications de différents contributeurs et de gérer les conflits de fusion.

3. Collaboration : GitHub facilite la collaboration entre développeurs. Vous pouvez travailler sur un projet avec d'autres personnes en clonant (copiant) un référentiel, en apportant des modifications locales et en les proposant pour qu'elles soient fusionnées dans le référentiel principal via des demandes de tirage (pull requests). Les discussions, les révisions de code et les commentaires peuvent être gérés directement sur la plateforme.

4. Hébergement de sites web : GitHub propose GitHub Pages, qui vous permet d'héberger des sites web statiques directement à partir de vos référentiels. Cela facilite la création et la diffusion de sites web personnels, de documentations de projets, de pages de démonstration, etc.

5. Collaboration ouverte : GitHub permet également aux développeurs de contribuer à des projets open source. Vous pouvez rechercher des projets qui vous intéressent, cloner les référentiels correspondants, apporter des modifications et soumettre des demandes de tirage pour proposer vos contributions à la communauté.

En résumé, GitHub est une plateforme qui facilite la collaboration, le partage et le contrôle de version pour les projets de développement logiciel. Que vous travailliez sur un projet personnel, en équipe ou que vous souhaitiez contribuer à des projets open source, GitHub offre de nombreuses fonctionnalités pour simplifier le processus de développement et de collaboration.

**Comment by HadhemiBM** [comment]
GitHub est une plateforme de développement collaboratif qui permet aux développeurs de travailler ensemble sur des projets de manière efficace et organisée. Créée en 2008, elle est principalement utilisée pour l'hébergement de projets de logiciels en utilisant le système de gestion de versions Git , donc  il permet aux développeurs de suivre les modifications apportées au code source, de collaborer de manière transparente, et de gérer plusieurs versions du projet. Chaque modification est enregistrée dans des "commits", qui peuvent être facilement consultés, comparés, et fusionnés.
